### PR TITLE
PIM-9738: move from command to job

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

When a locale is removed from a channel, the completeness of the products needs to be recomputed. This was done by a command. PR aims to transform the deprecated command into a job to be followed in the job tracker. Moreover, there was no test on it, so I added spec and e2e to be sure it works.

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [x] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php [hunk 1]
```diff
@@ -27,28 +30,32 @@ class ChannelLocaleSubscriber implements EventSubscriberInterface
...
+
+    /** @var IdentifiableObjectRepositoryInterface */
+    private $jobInstanceRepository;
+
     public function __construct(
         LocaleRepositoryInterface $repository,
         BulkSaverInterface $saver,
-        CommandLauncher $commandLauncher,
-        TokenStorageInterface $tokenStorage
+        JobLauncherInterface $jobLauncher,
```
Comment (nmarniesse) [inline src/Akeneo/Channel/Bundle/EventListener/ChannelLocaleSubscriber.php:48]: It's a BC break

File: tests/back/Integration/catalog/technical/jobs.yml [hunk 2]
```diff
@@ -188,3 +188,8 @@ jobs:
         alias:     remove_non_existing_product_values
         label:     Remove non existing product values
         type:      remove_non_existing_product_values
+    remove_completeness_for_channel_and_locale:
```
Comment (nmarniesse) [inline tests/back/Integration/catalog/technical/jobs.yml:191]: I think we should add the job permission in the EE (icecat + minimal + migration)

File: upgrades/schema/Version_4_0_20210720132650_remove_completeness_for_channel_and_locale.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Add the remove_completeness_for_channel_and_locale
+ */
+final class Version_4_0_20210720132650_remove_completeness_for_channel_and_locale extends AbstractMigration
+{
+    public function up(Schema $schema) : void
+    {
+        if ($this->jobExists('remove_non_existing_product_values')) {
```
Comment (nmarniesse) [inline upgrades/schema/Version_4_0_20210720132650_remove_completeness_for_channel_and_locale.php:15]:
```suggestion
        if ($this->jobExists('remove_completeness_for_channel_and_locale')) {
```

**Review by nmarniesse — APPROVED** [review]
GTM after fix

File: upgrades/schema/Version_4_0_20210720132650_remove_completeness_for_channel_and_locale.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
```
Comment (anaisbaunelemaire) [inline upgrades/schema/Version_4_0_20210720132650_remove_completeness_for_channel_and_locale.php:1]: We are not allowed to do migrations in already released major version :-/

**Review by rouliane — APPROVED** [review]

**Review by pierrejolly — APPROVED** [review]

**Pull request merged**